int32 sub_id
string text
